### PR TITLE
Feature/empty table can be read in with correct column headers as empty dataframe

### PR DESCRIPTION
**Pull request by heidikira**
Issue to be fixed:
When a startable with no data (ie, a table with no rows), was read it, an empty table with no columns was returned.  The table did not contain the correct columns and unit information.
This 'empty table' scenario could occur when reading in a startable when there were no results to output.  It would still be useful to read in the correct table structure.

Identification of the problem:
During the parsing of the table blocks, a dictionary of column names and data is built iteratively at the end of the `make_table_json_precursor()` method.  
First, an empty dictionary is initialized, and then columns and corresponding data are added as value-key pairs to the dictionary.
Therefore, if there is no data, then the initialized empty dictionary is returned, and there is no column information.

Proposed fix:
Initialize the dictionary to include the column names as keys in the dictionary with empty lists to be replaced if there is data.
Empty lists are needed to work correctly in the dataframe creation step.

### CODE REVIEW

**Comment by heidikira** [comment]
Added another test for reading in empty json.  Both paths where `make_table_json_precursor()` is used are now tested.

File: pdtable/test/io/test_read_excel.py [hunk 1]
```diff
@@ -67,3 +68,20 @@ def is_table_about_spelling(block_type: BlockType, block_name: str) -> bool:
     assert len(blocks) == 1
     table: Table = blocks[0][1]
     assert table.name == "spelling_numbers"
+
+
+def test_read_excel__empty_table():
```
Comment (jfcorbett) [inline pdtable/test/io/test_read_excel.py:73]: This is really testing the same thing as the new test for `read_csv()`above. It's just a different entry point for the function that was actually changed, namely `make_table_json_precursor()` called via `parse_blocks()`.

Maybe the right thing to do would be to add a single test for `parse_blocks()`? Rather than three separate ones for CSV, Excel, JSON, which in the end all test the same underlying thing repeatedly.

If so, then this would supersede my earlier comment about the implementation details of the read_csv() test.

**Review by jfcorbett — CHANGES_REQUESTED** [review]
Props for fixing this!
The tests should perhaps be rethought. Rather than three separate tests for CSV, Excel, JSON, which in the end all test the same underlying thing repeatedly; maybe only test that underlying thing? i.e. `parse_blocks()` which eventually calls the only function that was changed, `make_table_json_precursor()`. See existing `test_parse_blocks()`

File: pdtable/test/io/test_read_excel.py (hunk 1 above)
Comment (heidikira) [reply pdtable/test/io/test_read_excel.py:73]: I agree.  There should be two tests for `parse_blocks()` --> one for each of the differently factory methods `[make_table(), make_table_json_data()]`
And I should probably test that json with 'empty' values can be correctly turned into a corresponding dataframe.

Comment (heidikira) [reply pdtable/test/io/test_read_excel.py]: I have removed the excel and csv tests.

I added one test to test_block_parsers.py, namely` test_make_table__empty_values()`, which calls `make_table()` using data with no values and verifies the correct dataframe is created.

I modified the `test_make_table_json_data__empty_table()`  to directly call `make_table_json_data()`  and verify the correct json string is returned.  Furthermore, this json string with no values is also tested to ensure `json_data_to_table()` creates the correct dataframe.

**Review by jfcorbett — APPROVED** [review]
:shipit:
